### PR TITLE
Handle consent changes

### PR DESCRIPTION
**Pull request by mrotondo**
Add a Program model which stores opt-in/out keywords and responses in multiple languages, a ConsentChange model which records consent changes resulting from inbound texts, and a service class to process texts to identify and handle consent changes

Story ID #183621330

### CODE REVIEW

**Comment by mrotondo** [comment]
> I was thinking that this was all a violation of YAGNI, but am adjusting my thinking based on the Slack conversation today.

Yeah, I only added the per-program piece because Dan asked for it in a previous meeting. I could be convinced to remove it if we decide that because the prototype will only be used for SNAP renewal reminders, then any opt out during the pilot is technically a SNAP-related opt out, but in any case I found it helpful to think through the design.

File: db/migrate/20221110020049_make_recipient_case_id_optional.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class MakeRecipientCaseIdOptional < ActiveRecord::Migration[7.0]
+  def change
+
+    change_column_null :recipients, :message_batch_id, true
```
Comment (vrajmohan) [inline db/migrate/20221110020049_make_recipient_case_id_optional.rb:4]: Why is this needed?

Comment (mrotondo) [reply db/migrate/20221110020049_make_recipient_case_id_optional.rb:4]: It's a consequence of moving status out of SmsMessage and onto Recipient, which means that we now need a Recipient to properly send a message and keep track of its delivery. So now not all Recipients are part of a message batch: Some now only exist to send a response to a consent change message. I don't like the current polymorphic feel of Recipient - maybe we can think of something better.

File: app/services/consent_service.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+class ConsentService
+
+  def process_consent_change(sms_message)
+    return unless sms_message.direction == "inbound"
+    Program.all.each do |program|
+      Rails.logger.info sms_message.body
```
Comment (mrotondo) [inline app/services/consent_service.rb:6]: @mrotondo delete

File: db/seeds.rb [hunk 3]
```diff
@@ -5,3 +5,33 @@
...
+    en: %w(START UNSTOP STARTSNAP),
+    es: %w(COMENZAR COMENZARSNAP)},
+  opt_out_keywords: {
+    en: %w(STOP END CANCEL UNSUBSCRIBE QUIT STOPSNAP),
+    es: %w(DETENER FIN CANCELAR SUSCRIBIRSE DEJAR ALTOSNAP)},
+  opt_in_response: "Hello",
+  opt_out_response: "You have successfully unsubscribed from DSS reminders. You will not receive any more messages from this number. Reply START to resubscribe."
+)
+Mobility.with_locale(:es) do
+  snap.opt_in_response = "Hola"
```
Comment (vrajmohan) [inline db/seeds.rb:34]: Why is this set after creation as opposed to initialized for multiple locales like `opt_in_keywords` and `opt_out_keywords`?

Comment (mrotondo) [reply db/seeds.rb:36]: Made these consistent as discussed on zoom today

**Review by vrajmohan — APPROVED** [review]
